jenkins/image_building/initrd_sdk/unlock-mount-luks.sh

```diff
@@ -99,68 +159,300 @@ _dry_run() {
+    local _target_param _params
+    _target_param="${1:?}"
+
+    IFS=" " read -r -a _params < "/proc/cmdline"
+
+    for _param in "${_params[@]}"; do
+        _key="${_param%=*}"
+        _value="${_param#*=}"
+        if [[ "${_target_param}" == "${_key}" && "${_key}" != "${_value}" ]];\
+        then
+            printf "%s" "${_value}"
+            break
+        fi
+    done
+}
+
+_detect_part_label_kernel_param() {
+    # Detects GPT partitions based on the presence of kernel cmdline
+    # parameter defined in the key=value format where the value is the
+    # GPT partition label.
+    # Arguments:
+    # - the "key" part of the kernel command line argument
+    # - optional default partition label to look for
+    local _param _label _default_label _dev_path
+    _param=${1:?}
+    _default_label=${2:-}
+    _label="$(_get_kernel_param_value "${_param}")"
+    if [[ -z "${_label}" ]]; then
+        _label="${_default_label}"
+    fi
+    _dev_path="/dev/disk/by-partlabel/${_label}"
+    if [[ ! -r "${_dev_path}" ]]; then
+        return
+    fi
+    printf "%s" "${_dev_path}"
+}
+
+_detect_root() {
+    # This function checks the kernel command line variables and checks if the
+    # variable p.root_label is present, if present with a non empty value
+    # then the value will be return otherwise nothing will be returned.
+    # In case the label can't be found the function will default to looking
+    # for the p.lxroot label.
+    local _root_dev
+    _root_dev="$(_detect_part_label_kernel_param p.root_label p.lxroot)"
+    printf "%s" "${_root_dev}"
+}
+
+_detect_config() {
+    # This function checks the kernel command line variables and checks if the
+    # variable p.config_label is present, if present with a non empty
+    # value then the value will be return otherwise nothing will be returned.
+    # In case the label can't be found the function will default to looking
+    # for the p.config-2 label.
+    local _cfg_dev
+    _cfg_dev="$(_detect_part_label_kernel_param p.config_label p.config-2)"
+    printf "%s" "${_cfg_dev}"
+}
+
 # Start of the execution
 
-root_device_path="${1:?}"
-key_script="${2:-/etc/tpm2-unseal-key.sh}"
-config_drive_part_num="${3:-}"
-dry_run="${4:-false}"
-key="${key_script}"
+config="${1:-/etc/unlock_conf}"
+
+if [[ ! -r "${config}" ]];then
+    printf "WARNING: Executing script with default configuration!\n"
+else
+    printf "INFO: Executing script with configuration from:%s!\n" "${config}"
+    # shellcheck disable=SC1090
+    source "${config}"
+fi
+
+# Config variables, only configurable via the config file:
+# key management config
+key_script="${key_script:-/etc/tpm2-unseal-key.sh}"
+auth="${auth:-secret}"
+secret_address="${secret_address:-0x81010002}"
+# root partition config
+root_dev_part_path="${root_dev_part_path:-$(_detect_root)}"
+# config drive partition config
+config_drive_dev_path="${config_drive_dev_path:-$(_detect_config)}"
```

Comment:
Empty values are fine and expected, I have added an extra check for /proc/cmdline other that, these functions only access /dev/disk/by-partlabel and missing device is handled. No external tool is used only /proc and /dev fs access.

I might add some sanity checks for block devices in later PRs, but eventually blkid our mount would fail anyway and the paths are logged before that. Device files can have any names and can be located in many places in /dev .

I think I will just run a sanity check to see if these point to readable files or not that I can do.